### DISCUSSION TITLE
Getting error while using GSM module on LILYGO_T-SIM7600E/G-H

### DISCUSSION THREAD

**Discussion by userHarpreet** [mobizt/ESP-Google-Sheet-Client#34] [Q&A] [answered]
I am using GSM modem and trying to append data. Changed the config as per [usage-with-tinygsm](https://github.com/mobizt/ESP-Google-Sheet-Client?tab=readme-ov-file#usage-with-tinygsm) but getting the following error:

Token info: type = OAuth2.0 access token, status = on initializing
Token info: type = OAuth2.0 access token, status = on signing
Token info: type = OAuth2.0 access token, status = on exchange request
Token info: type = OAuth2.0 access token, status = error
Token error: code: -4, message: connection lost

Please Help.

**Comment by mobizt** [comment, marked as answer]
You have to follow this [example](https://github.com/mobizt/ESP-Google-Sheet-Client/blob/master/examples/GSM/GSM.ino).

The first important thing is you have to define this macro in [src/ESP_Google_Sheet_Client_FS_Config.h](https://github.com/mobizt/ESP-Google-Sheet-Client/blob/master/src/ESP_Google_Sheet_Client_FS_Config.h) in the library installation folder.
`#define TINY_GSM_MODEM_SIM7600`

And that macro also defined in your sketch too.

**Reply by userHarpreet** [reply]
Thanks for a prompt response, it was already there in the sketch but defining it in the config did the magic. Thanks again.
